### PR TITLE
Moodle401 fixes

### LINKED ISSUE #197: Deprecated code in renderer.php
Deprecated: Required parameter $config follows optional parameter $canedit in
local/envbar/renderer.php on line 254

### PR DESCRIPTION
**Pull request by sarahjcotton**
Fixes #197 and #199

### CODE REVIEW

File: renderer.php [hunk 1]
```diff
@@ -251,7 +251,7 @@ function envbar_close(el) {
      * @param object $config Config
      * @return string Debug text
      */
-    protected function get_debug_text($canedit = false, $config) {
+    protected function get_debug_text($config, $canedit = false) {
```
Comment (brendanheywood) [inline renderer.php]: was there a reason for swapping the order of these args?

Comment (sarahjcotton) [reply renderer.php]: Apparently optional params should come last in php 8+
https://www.php.net/manual/en/migration80.deprecated.php

Fixes:
Deprecated: Required parameter $config follows optional parameter $canedit in
local/envbar/renderer.php on line 254

Or would you prefer to drop the default value instead?

Comment (brendanheywood) [reply renderer.php]: Nope that makes perfect sense thanks

**Comment by brendanheywood** [comment]
@sarahjcotton this looks ok code wise but a few bits of house keeping:

- ci is still red for 3.10
- can you bump the version and supports metadata
- and it is using the old shared workflows. Are you able to convert that over as part of getting this green?

https://github.com/catalyst/catalyst-moodle-workflows#add-the-workflow

File: version.php [hunk 2]
```diff
@@ -29,9 +29,9 @@
...
-$plugin->version   = 2022011401;      // The current plugin version (Date: YYYYMMDDXX).
-$plugin->release   = 2022011401;      // Same as version
-$plugin->requires  = 2014051200;      // Requires Moodle 2.7 or later.
+$plugin->version   = 2023010900;      // The current plugin version (Date: YYYYMMDDXX).
+$plugin->release   = 2023010900;      // Same as version
+$plugin->requires  = 2022112800;      // Requires Moodle 2.7 or later.
 $plugin->component = 'local_envbar';
 $plugin->maturity  = MATURITY_STABLE;
-$plugin->supported = [310, 311];      // A range of branch numbers of supported moodle versions.
+$plugin->supported = [401, 401];      // A range of branch numbers of supported moodle versions.
```
Comment (dmitriim) [inline version.php]: I can't see any specific changes for 4.1 exclusively. Why are we dropping support for 310 and 311 here? And also bumping a min version in line 34?

**Comment by sarahjcotton** [comment]
I'm hoping this is ready to go now. I've updated the badge in the README and changed the ci workflow to hopefully the right one as well as fixed the ci issues.

**Pull request merged**